File: RetrofitJava/app/src/main/java/com/celalkorucu/retrofitjava/adapter/CryptoAdapter.java
package com.celalkorucu.retrofitjava.adapter;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.celalkorucu.retrofitjava.databinding.CryptoRecyclerRowBinding;
import com.celalkorucu.retrofitjava.model.CryptoModel;

import java.util.ArrayList;

public class CryptoAdapter extends RecyclerView.Adapter<CryptoAdapter.CryptoHolder> {


   ArrayList<CryptoModel> cryptoModelArrayList ;
   String colors [] = {"#92dfff","#00703c","#01796f","#177245","#8c92ac","#cd7f32","#009966","#009900","#dcdcdc","#5b264f"};
    public CryptoAdapter(ArrayList<CryptoModel>  cryptoModelArrayList){
        this.cryptoModelArrayList = cryptoModelArrayList ;


    }


    @NonNull
    @Override
    public CryptoHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        CryptoRecyclerRowBinding binding = CryptoRecyclerRowBinding.inflate(LayoutInflater.from(parent.getContext()) , parent , false);

        return new CryptoHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull CryptoHolder holder, int position) {
        holder.binding.cryptoName.setText(cryptoModelArrayList.get(position).getCurrency());
        holder.binding.cryptoPrice.setText(cryptoModelArrayList.get(position).getPrice()+" $");
        holder.bind(position);
    }



    @Override
    public int getItemCount() {
        return cryptoModelArrayList.size();
    }

    public class CryptoHolder extends RecyclerView.ViewHolder{

        CryptoRecyclerRowBinding binding ;

        public CryptoHolder(CryptoRecyclerRowBinding binding) {
            super(binding.getRoot());
            this.binding=binding ;
        }
        public void bind(int position){
            binding.getRoot().setBackgroundColor(Color.parseColor(colors[position%10]));
        }
    }
}

File: KotlinRetrofitAPI/app/src/main/java/com/celalkorucu/kotlinretrofitapi/CryptoAdapter.kt
package com.celalkorucu.kotlinretrofitapi

import android.graphics.Color
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.celalkorucu.kotlinretrofitapi.databinding.RecyclerRowBinding

class CryptoAdapter( private val cryptoList : ArrayList<CryptoModel> , private val listener : Listener) :
    RecyclerView.Adapter<CryptoAdapter.CryptoHolder>() {


    private val colors: Array<String> = arrayOf("#13bd27","#29c1e1","#b129e1","#d3df13","#f6bd0c","#a1fb93","#0d9de3","#ffe48f")


    class CryptoHolder( val binding: RecyclerRowBinding) :
        RecyclerView.ViewHolder(binding.root)



    interface Listener {
        fun onItemClick(cryptoModel: CryptoModel)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CryptoHolder {

        val binding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return CryptoHolder(binding)

    }
    override fun onBindViewHolder(holder: CryptoHolder, position: Int) {
        holder.itemView.setOnClickListener {
            listener.onItemClick(cryptoList.get(position))
        }
        holder.itemView.setBackgroundColor(Color.parseColor(colors[position % 8]))
        holder.binding.cryptoName.text = cryptoList.get(position).currency
        holder.binding.cryptoPrice.text = cryptoList.get(position).price
    }
    override fun getItemCount(): Int {

        return cryptoList.size
    }

}
